src/test/lib/operators/import_test.cpp

```diff
@@ -18,55 +27,215 @@ class OperatorsImportTest : public BaseTest {
+
+  auto segment = table->get_chunk(ChunkID{0})->get_segment(ColumnID{0});
+  auto expected_segment = expected_table->get_chunk(ChunkID{0})->get_segment(ColumnID{0});
+
+  // Check if the imported segment has the same type as the manually encoded one.
+  resolve_segment_type<float>(*segment, [&](const auto& segment) {
+    EXPECT_TRUE(std::dynamic_pointer_cast<std::remove_reference_t<decltype(segment)>>(expected_segment));
+  });
 }
 
-TEST_P(OperatorsImportMultiFileTypeTest, HasCorrectMvccData) {
+TEST_P(OperatorsImportMultiFileTypeAndEncodingTest, HasCorrectMvccData) {
+  const auto& [file_type, encoding] = GetParam();
+
   const auto reference_filename =
-      reference_filepath + reference_filenames.at(GetParam()) + file_extensions.at(GetParam());
-  auto importer = std::make_shared<Import>(reference_filename, "a");
+      reference_filepath + reference_filenames.at(file_type) + file_extensions.at(file_type);
+  auto importer = std::make_shared<Import>(reference_filename, "a", Chunk::DEFAULT_SIZE, file_type, encoding);
   importer->execute();
 
-  auto table = Hyrise::get().storage_manager.get_table("a");
+  const auto table = Hyrise::get().storage_manager.get_table("a");
 
   EXPECT_EQ(table->uses_mvcc(), UseMvcc::Yes);
   EXPECT_TRUE(table->get_chunk(ChunkID{0})->has_mvcc_data());
   EXPECT_EQ(table->get_chunk(ChunkID{0})->mvcc_data()->max_begin_cid.load(), CommitID{0});
 }
 
+TEST_F(OperatorsImportTest, EncodeImportWithFrameOfReference) {
+  const auto& encoding_type = EncodingType::FrameOfReference;
+  auto expected_table =
+      std::make_shared<Table>(TableColumnDefinitions{{"a", DataType::Int, false}}, TableType::Data, ChunkOffset{4});
+
+  expected_table->append({123});
+  expected_table->append({1234});
+  expected_table->append({12345});
+
+  // Make all chunks in the expected table immutable to allow encoding.
+  for (auto chunk_id = ChunkID{0}; chunk_id < expected_table->chunk_count(); ++chunk_id) {
+    expected_table->get_chunk(chunk_id)->set_immutable();
+  }
+  // Encode the expected table with FrameOfReference.
+  const auto encoding_spec = SegmentEncodingSpec{encoding_type};
+  ChunkEncoder::encode_all_chunks(expected_table, encoding_spec);
+
+  const auto filename = std::string{"resources/test_data/tbl/int.tbl"};
+  auto importer = std::make_shared<Import>(filename, "a", Chunk::DEFAULT_SIZE, FileType::Tbl, encoding_type);
+  importer->execute();
+
+  const auto table = Hyrise::get().storage_manager.get_table("a");
+
+  EXPECT_TABLE_EQ_ORDERED(table, expected_table);
+
+  auto segment = table->get_chunk(ChunkID{0})->get_segment(ColumnID{0});
+  auto expected_segment = expected_table->get_chunk(ChunkID{0})->get_segment(ColumnID{0});
+
+  // Check if the imported segment has the same type as the manually encoded one.
+  resolve_segment_type<float>(*segment, [&](const auto& segment) {
+    EXPECT_TRUE(std::dynamic_pointer_cast<std::remove_reference_t<decltype(segment)>>(expected_segment));
+  });
```

Comment:
you know you want a `FrameOfReferenceSegment<int32_t>`:
```suggestion
  EXPECT_TRUE(std::dynamic_pointer_cast<const FrameOfReferenceSegment<int32_t>>(expected_segment));
```